code/modules/mob/mob.dm

```diff
@@ -28,6 +28,9 @@
 		GLOB.living_mob_list += src
 	prepare_huds()
 	can_ride_typecache = typecacheof(can_ride_typecache)
+	for(var/v in GLOB.active_alternate_appearances)
+		var/datum/atom_hud/alternate_appearance/AA = v
+		AA.onNewMob(src)
```

Comment:
Looping through this list every time a mob is spawned is not great, but it solves the problem.